### DISCUSSION TITLE
Connection Unstable Issue while Running Module 1.4 of Self-Driving Laboratories

### DISCUSSION THREAD

**Discussion by greenmossball** [AccelerationConsortium/ac-microcourses#133] [Hello World]
Hi all,

I'm experiencing an error message when running the mqtt_led.py script described in the 1.4 module (Device communication). This is the error message I see:

Obtaining CA Certificate
Checking WiFi integrity.
Traceback (most recent call last):
  File "<stdin>", line 119, in <module>
  File "asyncio/core.py", line 1, in run
  File "asyncio/core.py", line 1, in run_until_complete
  File "asyncio/core.py", line 1, in run_until_complete
  File "<stdin>", line 102, in main
  File "mqtt_as.py", line 654, in connect
  File "mqtt_as.py", line 646, in wifi_connect
OSError: Connection Unstable

Any help will be greatly appreciated. I was able to run the "Run the demo" module so I'm not sure what I'm doing wrong.

**Comment by sgbaird** [comment]
I don't think I've seen a "connection unstable" error before. Could you try with one or two other WiFi networks? (E.g., phone hotspot).

Did you finish module 1 recently? Might be worth going back to check to make sure it still runs OK. May help with troubleshooting.

**Comment by greenmossball** [comment]
Hi Sterling,

Yes, I tried it again just to see if something might have changed between
then and now. It seems to run fine. I'll try running in another Wifi. The
phone hotspot doesn't seem to work,btw.

best,
-a-

**Reply by sgbaird** [reply]
Thanks!

> The phone hotspot doesn't seem to work

Oof. I'll ask someone to see if they can reproduce it. Do you mind also opening an issue on the mqtt_as repo? They'd probably know about this error better than me.

I'll keep digging.

**Reply by sgbaird** [reply]
@RubainaFarin31 could you test out the mentioned code to see if you run into the same error or not?

**Reply by sgbaird** [reply]
@greenmossball any update with trying another WiFi?

**Comment by greenmossball** [comment]
Hi Sterling,

I tried wifi from my phone but that did not work either. I just answered
the questions of the tests as best as i could and moved on and completed
the course.

best,
-a-

**Reply by sgbaird** [reply]
Ah, OK. Sorry to hear that. One key difference is that `self-driving-lab-demo` package uses `mqtt.simple`, whereas module 1.4 uses `mqtt_as` (typically more robust and offers more features). I've been meaning to switch `self-driving-lab-demo` over to `mqtt_as`, but left it as-is since it's been working fine. I'll open an issue about the instability on `mqtt_as` repo.

**Reply by sgbaird** [reply]
One possible workaround might be to replace `await client.connect()` with `await client.connect(quick=True)`
